### PR TITLE
Removes the legacy-action Integrity Check

### PR DESCRIPTION
**Pull request by simonharrer**
Removes the **legacy**-action `Tools--> Legacy Tools... --> Integrity Check` with all its files. 

What it helps:
- removes 1000 lines of code
- removes a feature that is marked as a legacy tool

What does the feature do at the moment:
- check year: raise error if it contains a 4digit number
- check title: warn only about wrong braces
- check author/editor: check that the structure is correct using `and` and the `,` separator.

Anyone who wants this feature to stay?

Is there any other validation facility in place where we could add the checks instead?

### CODE REVIEW

**Comment by koppor** [comment]
Wow, the features look impressive!

Does it check for presence or absence of 4 digit year number? I think, presence of 4 digits makes sense?

Wrong braces: Maybe we should have two modes for a bibtex file - collecting and paper writing (see  #113). And for paper writing, we should have some brace check. In the collecting mode, no braces should be used at all.

Author/editor check: What a useful tool. I'm always struggling with that.

We should keep these checks! The checks should be **moved** to the fields: If a field looses the focus, the check should be fired. If the field is wrong, the field should be colored in red.

**Comment by simonharrer** [comment]
Is there already a similar check for another field in place? Or should we
just Mark the Entry red in the table as when there is something missing?
Am 09.09.2015 8:06 nachm. schrieb "Oliver Kopp" <notifications@github.com>:

**Comment by matthiasgeiger** [comment]
At least the BibTex-Key is currently checked (e.g., spaces are not allowed).

**Comment by matthiasgeiger** [comment]
There is already a related feature discussion here: #131

**Comment by koppor** [comment]
I even didn't know the check for the BibTex key. JabRef really insists on correct BibTeX keys there...

An entry as a whole is marked in red if something is wrong: The number in the first column gets a red background. I suppose, it is only marked if a required field is missing. In the case of `article` additionally, if `volume` and `pages` is missing. But there is no fly-over-help at the red number :worried:

**Comment by simonharrer** [comment]
Hm, this is a lot of effort to put a system in place which can handle such checks and validations in a sensible way. 

Maybe we can leverage the tool tip text that is set in the IncompleteRenderer. But I cannot see the tool tip text in JabRef ...

**Comment by koppor** [comment]
Is it an aspect for #113? Maybe we should close this pull request and add a new issue to implement the checking behavior. JabRef should support the user whenever possible :smiley:

**Comment by simonharrer** [comment]
Large UPDATE:

- [x] refactored existing integrity checks and provide test cases for them
- [x] instead of complex GUI elements, use a simple results table for displaying the validation results
- [x] instead of having separate language property files for a few keys, use the default ones
- [x] minimal but working example of such a validation

I propose to merge this as this has more value than the "legacy" implementation before.

File: src/main/java/net/sf/jabref/logic/l10n/Localization.java [hunk 1]
```diff
@@ -50,7 +46,7 @@ public static void setLanguage(String language) {
      *
      * @param resBundle the ResourceBundle to use
      * @param idForErrorMessage output when translation is not found
-     * @param key the key to lookup in resBundle
+ö     * @param key the key to lookup in resBundle
```
Comment (matthiasgeiger) [inline src/main/java/net/sf/jabref/logic/l10n/Localization.java]: :see_no_evil:

**Comment by matthiasgeiger** [comment]
Great, that this integrity message resource properties are gone now ;-) 

And for the first time this feature seems to work :wink:

Not sure how much time it requires to implement, but a nice feature would be to open the affected entry and setting the cursor in the affected field when selecting a found integrity issue. 

Another minor aspect: the results table currently wastes a lot of space for the affected field but the message is not completely readible.

**Comment by simonharrer** [comment]
Good ideas. But I think the current state is good enough to have a usable and helpful feature.

**Comment by matthiasgeiger** [comment]
I think the aspects above are rather crucial for good usability - I will check how easy this can be implemented...

**Comment by matthiasgeiger** [comment]
Done. :metal:

**Comment by simonharrer** [comment]
:trophy: awesome.

**Comment by simonharrer** [comment]
Bug: when the field for which the validation files is not required, the field is not found. 

Example: title has an error, but the entry is of type Misc and only requires a bibtex key.

**Comment by matthiasgeiger** [comment]
Fixed in f9ea1d1fe1dd5d01457fecfabbfe522da9e4cfd3

**Pull request merged**